fedimint-client-wasm/src/lib.rs

```diff
@@ -0,0 +1,134 @@
+        let invite_code = InviteCode::from_str(&invite_code)?;
+        let config = fedimint_api_client::download_from_invite_code(&invite_code).await?;
+        let client = Arc::new(builder.join(root_secret, config, None).await?);
+        Ok(Self { client })
+    }
+
+    #[wasm_bindgen]
+    /// Call a fedimint client rpc the responses are returned using `cb`
+    /// callback. Each rpc call *can* return multiple responses by calling
+    /// `cb` multiple times. You should ignore the promise by this function
+    /// because it has no significance.
+    pub async fn rpc(&self, module: &str, method: &str, payload: String, cb: &js_sys::Function) {
+        let mut stream = pin!(self.rpc_inner(module, method, payload));
+
+        while let Some(item) = stream.next().await {
+            let this = JsValue::null();
+            let _ = match item {
+                Ok(item) => cb.call1(
+                    &this,
+                    &JsValue::from_str(&serde_json::to_string(&item).unwrap()),
+                ),
+                Err(err) => cb.call1(
+                    &this,
+                    &JsValue::from_str(
+                        &serde_json::to_string(&json!({"error": err.to_string()})).unwrap(),
+                    ),
+                ),
+            };
+        }
+    }
+
+    fn rpc_inner<'a>(
+        &'a self,
+        module: &'a str,
+        method: &'a str,
+        payload: String,
+    ) -> impl futures::Stream<Item = anyhow::Result<serde_json::Value>> + 'a {
+        try_stream! {
+            let payload: serde_json::Value = serde_json::from_str(&payload)?;
+            match module {
+                "" => {
+                    let mut stream = self.client.handle_global_rpc(method.to_owned(), payload);
+                    while let Some(item) = stream.next().await {
+                        yield item?;
+                    }
+                }
+                "ln" => {
+                    let ln = self
+                        .client
+                        .get_first_module::<LightningClientModule>()
+                        .inner();
+                    let mut stream = ln.handle_rpc(method.to_owned(), payload).await;
+                    while let Some(item) = stream.next().await {
+                        yield item?;
+                    }
+                }
+                "mint" => {
+                    let mint = self
+                        .client
+                        .get_first_module::<fedimint_mint_client::MintClientModule>()
+                        .inner();
+                    let mut stream = mint.handle_rpc(method.to_owned(), payload).await;
+                    while let Some(item) = stream.next().await {
+                        yield item?;
+                    }
+                }
+                _ => {
+                    Err(anyhow::format_err!("module not found: {module}"))?;
+                    unreachable!()
+                },
+            }
```

Comment:
Does this need to be matched explicitly? Shouldn't every module know its kind and we can just search for the `DynClientModule` with the correct kind and then call `handle_rpc` on it?

That way it would be much easier to make the `fedimint-client-wasm` crate instantiable with different module sets like all the other top-level crates.

Reply:
:+1: will do in follow up